docs/ai/conceptual/understanding-tokens.md

```diff
@@ -0,0 +1,118 @@
+| Larger tokens (word tokenization)                  | - A given text is broken into fewer tokens, requiring less computational resources while processing<br>- Given the same token limit, the maximum size of the model's input and output is larger | - Might cause an increased vocabulary size, requiring more memory resources<br>- Can limit the models ability to handle unknown words, typos, or complex syntax                                      |
+
+## How LLMs use tokens
+
+After completing tokenization, LLMs assign an ID to each unique token.
+
+Consider the following text:
+
+> I heard a dog bark loudly at a cat
+
+After using a word tokenization method, a model could assign token IDs like this:
+
+- I (1)
+- heard (2)
+- a (3)
+- dog (4)
+- bark (5)
+- loudly (6)
+- at (7)
+- a (the "a" token is already assigned an ID of 3)
+- cat (8)
+
+By assigning IDs, text can be represented as a sequence of token IDs. The previous sentence would be represented as [1, 2, 3, 4, 5, 6, 7, 3, 8]. The sentence "I heard a cat" would be represented as [1, 2, 3, 8].
+
+As training continues, the model add any new tokens in the training text to it's vocabulary and assigns it an ID. For example:
+
+- meow (9)
+- run (10)
+
+The semantic relationships between the tokens can be analyzed using these token ID sequences. Multi-valued numeric vectors, known as [_embeddings_]<!-- (understanding-embeddings.md) -->, are used to represent these relationships. An embedding is assigned to each token based on how commonly it's used together with or in similar contexts to the other tokens.
+
+Once trained, a model can calculate an embedding for a text containing multiple tokens. The model tokenizes the text, then calculates an overall embeddings value based on the learned embeddings of the individual tokens. This technique can be used for semantic document searches or adding [memories](/semantic-kernel/memories/) to an AI.
+
+During output generation, the model predicts a vector value for the next token in the sequence. The model then selects the next token from it's vocabulary based on this vector value. In practice, the models calculates multiple vectors using various elements of the preceding tokens' embeddings. The model then evaluates all potential tokens from these vectors and selects the most probable one to continue the sequence.
+
+Output generation is an iterative operation. The model appends the predicted token to the sequence so far and uses that as the input for the next iteration, building the final output one token at a time.
+
+### Token limits
+
+LLMs have limits on the maximum number of tokens that can be used as input or generated as output. This limit often combines the input and output tokens into a maximum context window.
```

Comment:
```suggestion
LLMs have limitations regarding the maximum number of tokens that can be used as input or generated as output. This limitation often causes the input and output tokens to be combined into a maximum context window.
```
You don’t have limits, you have limitations (inherent). A regulatory agency imposes limits; you obey them. If the LLM is imposing the limits, we can use, "LLMs put limits on the..." and we can leave the "This limit often combines..." sentence as is.